packages/skill-api/src/core/services/process-stripe-webhook.ts

```diff
@@ -92,14 +96,12 @@ export async function processStripeWebhooks({
         if (user) {
           const currentEmail = user.email
           const {email, name} = event.data.object
-          await prisma.user.update({
-            where: {
-              id: user.id,
-            },
-            data: {
-              email,
-              name,
-            },
```

Comment:
the issue was that in Total TypeScript you can have an account using the target email already where in Testing accessibility you couldn't have an account without a purchase. This TRANSFERS the purchase instead of updating the user.

Reply:
The individual pieces of this look good. I'm getting hung up on understanding the scenario this is for. I'll try to spell out what I think the scenario is, though it is mostly questions checking for understanding.

This is to solve an issue with users and purchases in Total TypeScript, right? (as opposed to TA)

A person ends up with two separate `User` records? And we need to transfer any `Purchase` and `MerchantCharge` records from one to the other? When would that happen?

It seems like because of line 100, we are either looking up the one `User` record for this person's email or creating that initial `User` record. How is it that they already have purchases at this point and from what `User` record are we transferring them?

Or is it that there are existing `Purchase` and `MerchantCharge` records for which `userId` is `null`?

Reply:
@jbranchaud with TT we allow you to login and create an account. This means that when I do the email change in stripe the previous logic failed because it can't update the purchasing user's email to the new email.

TA does NOT allow login without purchase, but this would work fine there as well.

Downside: progress is lost.

Instead of a mutation, we treat user's email addresses as immutable and instead transfer the purchase to a new user.

This will be similar to the self-serve "transfer purchase" functionality that we will be adding, and could probably be directly used for that.